testsuite/tests/basic/equality.ml

```diff
@@ -72,35 +72,78 @@ let _ =
   test 36 eqm1 (compare (0.0, nan) (0.0, 0.0));
   test 37 eqm1 (compare (0.0, nan) (0.0, neg_infinity));
   test 38 eq0 (compare (nan, 0.0) (nan, 0.0));
-  let cmpgen x y = (x=y, x<>y, x<y, x<=y, x>y, x>=y) in
-  let cmpfloat (x:float) (y:float) = (x=y, x<>y, x<y, x<=y, x>y, x>=y) in
+
+  (* Force inline these functions to ensures that the simplifier will
+     simplify the comparisons. *)
+  let [@inline always] cmpgen x y =
+    (x=y, x<>y, x<y, x<=y, x>y, x>=y)
+  in
+  let [@inline always] cmpfloat (x:float) (y:float) =
+    (x=y, x<>y, x<y, x<=y, x>y, x>=y)
+  in
   test 39 eqftffff (cmpgen nan nan);
   test 40 eqftffff (cmpgen nan 0.0);
   test 41 eqftffff (cmpfloat nan nan);
   test 42 eqftffff (cmpfloat nan 0.0);
   test 43 eqtrue ([||] = [||]);
   (* Convoluted forms to test both the "positive" and "negative" cases
      of float tests *)
-  let cmpfloatpos (x:float) (y:float) =
+  let [@inline always] cmpfloatpos (x:float) (y:float) =
     ((let r = ref false in (if x = y then r := true); !r),
      (let r = ref false in (if x <> y then r := true); !r),
      (let r = ref false in (if x < y then r := true); !r),
      (let r = ref false in (if x <= y then r := true); !r),
      (let r = ref false in (if x > y then r := true); !r),
      (let r = ref false in (if x >= y then r := true); !r))
-  and cmpfloatneg (x:float) (y:float) =
+  and [@inline always] cmpfloatneg (x:float) (y:float) =
     ((let r = ref true in (if not (x = y) then r := false); !r),
      (let r = ref true in (if not (x <> y) then r := false); !r),
      (let r = ref true in (if not (x < y) then r := false); !r),
      (let r = ref true in (if not (x <= y) then r := false); !r),
      (let r = ref true in (if not (x > y) then r := false); !r),
      (let r = ref true in (if not (x >= y) then r := false); !r)) in
-  let testcmpfloat x y =
+  let [@inline always] testcmpfloat x y =
     cmpfloatpos x y = cmpgen x y &&
     cmpfloatneg x y = cmpgen x y in
   test 50 eqtrue (testcmpfloat nan nan);
   test 51 eqtrue (testcmpfloat nan 0.0);
   test 52 eqtrue (testcmpfloat 0.0 nan);
   test 53 eqtrue (testcmpfloat 0.0 0.0);
   test 54 eqtrue (testcmpfloat 1.0 0.0);
-  test 55 eqtrue (testcmpfloat 0.0 1.0)
+  test 55 eqtrue (testcmpfloat 0.0 1.0);
+
+  (* Check the equality at run time. *)
```

Comment:
I would say "Prevent inlining of these functions to force evaluation of the comparisons at runtime and test the corresponding code generation".